src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,88 @@
+import classNames from 'classnames';
+import { Link, useSearchParams } from 'react-router-dom';
+type Sex = '' | 'female' | 'male';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+
+  const centuriesChange = (c: string, e: React.MouseEvent) => {
+    e.preventDefault();
+    const params = new URLSearchParams(searchParams);
+
+    const newCenturies = centuries.includes(c)
+      ? centuries.filter(centuri => centuri !== c)
+      : [...centuries, c];
+
+    params.delete('centuries');
+    newCenturies.forEach(centuri => params.append('centuries', centuri));
+    setSearchParams(params);
+  };
+
+  const handleQueryChange = (value: string) => {
+    const params = new URLSearchParams(searchParams);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+    value !== '' ? params.set('query', value) : params.delete('query');
+    setSearchParams(params);
+  };
+
+  const handleSexChange = (value: Sex, e: React.MouseEvent) => {
+    e.preventDefault();
+    const params = new URLSearchParams(searchParams);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+    value !== '' ? params.set('sex', value) : params.delete('sex');
+    setSearchParams(params);
+  };
+
+  const resetFilter = () => {
+    searchParams.delete('centuries');
+    searchParams.delete('sex');
+    searchParams.delete('query');
```

Comment:
The `resetFilter` function deletes the search parameters but does not update the URL with `setSearchParams`. Ensure you call `setSearchParams(searchParams)` after deleting the parameters to update the URL accordingly.